RecoVertex/PrimaryVertexProducer/src/DAClusterizerInZ_vect.cc

```diff
@@ -473,251 +829,225 @@ bool DAClusterizerInZ_vect::split(const double beta, track_t& tks, vertex_t& y,
         }
       }
     }
+#endif
 
     // split if the new subclusters are significantly separated
     if ((z2 - z1) > epsilon) {
       split = true;
-      double pk1 = p1 * y._pk[k] / (p1 + p2);
-      double pk2 = p2 * y._pk[k] / (p1 + p2);
-      y._z[k] = z2;
-      y._pk[k] = pk2;
-      y.InsertItem(k, z1, pk1);
+      double pk1 = p1 * y.pk_ptr[k] / (p1 + p2);
+      double pk2 = p2 * y.pk_ptr[k] / (p1 + p2);
+      y.z_ptr[k] = z2;
+      y.pk_ptr[k] = pk2;
+      y.insertItem(k, z1, pk1, tks);
+      if (k == 0)
+        y.extractRaw();
+
       nv++;
 
       // adjust remaining pointers
       for (unsigned int jc = ic; jc < critical.size(); jc++) {
-        if (critical[jc].second > k) {
+        if (critical[jc].second >= k) {
```

Comment:
This looks like a bug fix: does it have any sizeable effect?

Reply:
Yes, it's a bug. It was apparently found when ZT clustering was derived from Z clustering and never backported. Ithink it only has an effect when two clusters in the "critical" list (split candidates) are direct neighbours. Even if this goes wrong, it is very likely to be corrected in the next split/merge cycle. I have not directly looked for effects of this, but since the overall differences are small, I assume that this bug has a negligible effect.